Mejoras en capa_aplicacion y capa_persistencia

FreddyCH8 opened the pull request:
Se ha modificado el formato del `ProjectGuid` en `capa_aplicacion.csproj` y se han reordenado las referencias de ensamblados. Se han añadido referencias a `Servicios\ReporteService.cs` y a otros proyectos (`capa_dominio` y `capa_persistencia`).

En `ReporteNomina.cs`, se han introducido los métodos `SafeGet` y `SplitFullName` para mejorar la obtención de datos de `SqlDataReader`. Se ha actualizado la lógica de creación de objetos `Trabajador` y `Cargo`, y se ha añadido el objeto `Periodo`.

Se ha creado `ReporteService.cs` para gestionar consultas y listados de nóminas y periodos, incluyendo manejo de excepciones personalizadas.

Commit:
Mejoras en capa_aplicacion y capa_persistencia

Se ha modificado el formato del `ProjectGuid` en `capa_aplicacion.csproj` y se han reordenado las referencias de ensamblados. Se han añadido referencias a `Servicios\ReporteService.cs` y a otros proyectos (`capa_dominio` y `capa_persistencia`).

En `ReporteNomina.cs`, se han introducido los métodos `SafeGet` y `SplitFullName` para mejorar la obtención de datos de `SqlDataReader`. Se ha actualizado la lógica de creación de objetos `Trabajador` y `Cargo`, y se ha añadido el objeto `Periodo`.

Se ha creado `ReporteService.cs` para gestionar consultas y listados de nóminas y periodos, incluyendo manejo de excepciones personalizadas.

Jonathan03R commented on capa_persistencia/modulo_principal/ReporteNomina.cs:
```diff
@@ -20,6 +20,41 @@ public ReporteNomina()
 
+        // Se agrego esto porque Copilot dijo que esto unia el nombre completo
+        // en nombres y apellidos de forma segura
+        private static string SafeGet(SqlDataReader dr, string columnName)
```
esta en ingles que es SafGet? , por que pide como paramentros SqlDataReader

Jonathan03R replied:
Esta función no une nada ni separa nombres.
Su único trabajo es leer un valor de una columna de una fila SQL estas creando un helper..

Jonathan03R commented on capa_persistencia/modulo_principal/ReporteNomina.cs:
```diff
@@ -20,6 +20,41 @@ public ReporteNomina()
+            }
+        }
+
+        private static void SplitFullName(string fullName, out string nombres, out string apellidos)
```
esta mal quieres oncatenar nombre  solo es esto en la clase 

:public class Empleado
{
    public string Nombres { get; set; }
    public string Apellidos { get; set; }

    public string NombreCompleto => $"{Nombres} {Apellidos}".Trim();
}

necesitas concatenar el nombre NO, si en caso se necesita mostrar el nombre completo en un solo lugar que esa informacion se maneje desde el front @FreddyCH8

Jonathan03R commented on capa_persistencia/modulo_principal/ReporteNomina.cs:
```diff
@@ -35,34 +70,43 @@ public List<DetalleNomina> ConsultarNominaPorPeriodo(int periodoId, int? cargoId
                     while (dr.Read())
                     {
-                        
+                        // Intentamos obtener nombres y apellidos por separado; si no existen, separamos NombreCompleto.
```
estas que ifiere la IA  la ia no sabe que informacion va a llegar pero tu si , tu hiciste el proc que tipo de dato llega , esta llegando nombre completo o separado ?  eso depende de ti , aun asi tirene quie ver mucho el modelo como le estas pasando el nombre

Jonathan03R reviewed (changes requested):
revisalo @FreddyCH8  estas que te ahogas en una funcionalidad inecesaria , solo preocupare de traer la informacion tal cual estas traendo en el procedimiento almacenado. y dale al objeto que haga su trabajo ,

Commit:
- Se han Añadido cambios en ReporteService.cs añadiendo un debug para ver que errores tiene.
- Se han Añadido cambios en ReporteNomina.cs con los comentarios que han sido revisados y sugeridos por el Scrum Master (Jonathan). Ademas de comprobar por medio de debugs cual fue el problema de conectarse al procedimiento almacenado, ya se vio el problema y se corrigio respectivamente.
- Se ha agregado un nuevo DTO ReporteNominaDTO.cs para llamar los parametros más facil y rapidos para el ReporteNomina.
- La conexion ha sido cambiada al servidor ya creado por Jonathan donde se encuentran las tablas, procedimientos almacenados y inserciones.
- Se quito una referencia de Bonos.cs que quedo pendiente, en especifico la de Microsoft.Data.SQLClient que se dijo que eso se quitara para todos.
- No hubo tantas modificaciones en GenerarReporteController.cs, pero si en GenerarReporte.cshtml ya acomodado al ReporteNomina (aun falta añadir cosas a la vista, pero lo importante es ver que funcione el procedimiento almacenado)
- los jquery y bootstrap son añadidos para que funcione correctamente el Framework.

Se espera la revision correspondiente

FreddyCH8 commented:
- Se han Añadido cambios en ReporteService.cs añadiendo un debug para…
… ver que errores tiene.

- Se han Añadido cambios en ReporteNomina.cs con los comentarios que han sido revisados y sugeridos por el Scrum Master (Jonathan). Ademas de comprobar por medio de debugs cual fue el problema de conectarse al procedimiento almacenado, ya se vio el problema y se corrigio respectivamente.
- Se ha agregado un nuevo DTO ReporteNominaDTO.cs para llamar los parametros más facil y rapidos para el ReporteNomina.
- La conexion ha sido cambiada al servidor ya creado por Jonathan donde se encuentran las tablas, procedimientos almacenados y inserciones.
- Se quito una referencia de Bonos.cs que quedo pendiente, en especifico la de Microsoft.Data.SQLClient que se dijo que eso se quitara para todos.
- No hubo tantas modificaciones en GenerarReporteController.cs, pero si en GenerarReporte.cshtml ya acomodado al ReporteNomina (aun falta añadir cosas a la vista, pero lo importante es ver que funcione el procedimiento almacenado)
- los jquery y bootstrap son añadidos para que funcione correctamente el Framework.

Jonathan03R commented on capa_aplicacion/Servicios/ReporteService.cs:
```diff
@@ -19,24 +20,27 @@ public ReporteService()
         }
 
-        public List<DetalleNomina> ConsultarNominaPorPeriodo(int periodoId, int? cargoId = null)
+        public List<ReporteNominaDTO> ConsultarNominaPorPeriodo(int periodoId, int? cargoId = null)
```
No estas abriendo la conexion

Jonathan03R commented on capa_persistencia/modulo_principal/ReporteNomina.cs:
```diff
@@ -149,7 +115,7 @@ public List<DetalleNomina> ConsultarNominaPorPeriodo(int periodoId, int? cargoId
                 conexion.CerrarConexion();
```
en a capa de persistencia no se deberia ni abrir ni cerrar conexion

Jonathan03R reviewed (commented):
Has ese ultimo cambio y mandas @FreddyCH8

Commit:
- En ReporteService se agrego el metodo para AbrirConexion porque en la capa aplicacion se debe de realizar ese metodo.
- En ReporteNomina se quito el metodo de AbrirConexion ya que solo se encarga de los procedimientos almacenados.

Jonathan03R committed:
Merge branch 'develop' into feature/HU3Servicios
